From ee1a364f0a6dd44774a419b97526d2bac16d9017 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Thu, 12 May 2022 14:13:24 +0800
Subject: [PATCH] [riscv64] Implement relaxed_i16x8_q15mulr_s

Port commit fe443a4e1fd0e738fe1a1d51e80d48d6d4cc4c67

Bug: v8:12284

Change-Id: I19ba180ecc7cd0e5a3b6e5009cc2b8c40074399a
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3642243
Reviewed-by: ji qiu <qiuji@iscas.ac.cn>
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Commit-Queue: ji qiu <qiuji@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#80507}
---
 src/wasm/baseline/riscv64/liftoff-assembler-riscv64.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/wasm/baseline/riscv64/liftoff-assembler-riscv64.h b/src/wasm/baseline/riscv64/liftoff-assembler-riscv64.h
index 6d84bf5490b..2d842aba10b 100644
--- a/src/wasm/baseline/riscv64/liftoff-assembler-riscv64.h
+++ b/src/wasm/baseline/riscv64/liftoff-assembler-riscv64.h
@@ -2167,6 +2167,13 @@ void LiftoffAssembler::emit_i16x8_q15mulr_sat_s(LiftoffRegister dst,
   vsmul_vv(dst.fp().toV(), src1.fp().toV(), src2.fp().toV());
 }
 
+void LiftoffAssembler::emit_relaxed_i16x8_q15mulr_s(LiftoffRegister dst,
+                                                    LiftoffRegister src1,
+                                                    LiftoffRegister src2) {
+  VU.set(kScratchReg, E16, m1);
+  vsmul_vv(dst.fp().toV(), src1.fp().toV(), src2.fp().toV());
+}
+
 void LiftoffAssembler::emit_i64x2_bitmask(LiftoffRegister dst,
                                           LiftoffRegister src) {
   VU.set(kScratchReg, E64, m1);
-- 
2.35.1

